src/Microsoft.IdentityModel.Tokens/X509SecurityKey.cs

```diff
@@ -147,5 +147,27 @@ public X509Certificate2 Certificate
         {
             get { return _certificate; }
         }
+
+        /// <summary>
+        /// Returns a bool indicating if this key is equivalent to another key.
+        /// </summary>
+        /// <return>true if the keys are equal; otherwise, false.</return>
+        public override bool Equals(object obj)
+        {
+            X509SecurityKey other = obj as X509SecurityKey;
+            if (other == null)
+                return false;
+
+            return other.Certificate.ToString() == this._certificate.ToString();
```

Comment:
X509Certificate2.ToString() is a pretty heavy call. Would checking the Thumbprint suffice? Why do we introduce this method?